modin/numpy/arr.py

```diff
@@ -137,9 +137,20 @@ def __init__(
         _query_compiler=None,
         _ndim=None,
     ):
+        self._siblings = []
         ErrorMessage.single_warning(
             "Using Modin's new NumPy API. To convert from a Modin object to a NumPy array, either turn off the ExperimentalNumPyAPI flag, or use `modin.utils.to_numpy`."
         )
+        if isinstance(object, array):
+            _query_compiler = object._query_compiler.copy()
+            if not copy:
+                object._add_sibling(self)
+            _ndim = object._ndim
+        if isinstance(object, (pd.DataFrame, pd.Series)):
```

Comment:
```suggestion
        elif isinstance(object, (pd.DataFrame, pd.Series)):
```
super-small perf speedup 😆